### PR TITLE
[UI] Tensorboard support for multi user

### PR DESCRIPTION
**Pull request by Bobgy**
/area frontend
/assign @chensun 
/kind feature

Part of https://github.com/kubeflow/pipelines/issues/3294
UI image: gcr.io/gongyuan-pipeline-test/dev/frontend@sha256:de35d56815503c37789150a13d0e724e2b681fbf748c679d5971aa0834fb0369

Feature Changes:
* UI gets namespace from run and sends to node server tensorboard apis
* Node server requires namespace argument for tensorboard apis

Engprod Changes:
* Allow UI node server local development to access minio artifact api in cluster
* Wrote missing tests for tensorboard apis in node server
* Fixed url query decoded twice problem and added a unit test for it:  https://github.com/kubeflow/pipelines/pull/3355/files#diff-bbd9ad97d5d99d826328e300b108a070R690-R694

Verification:
* Tested on https://kfpmu-3-23.endpoints.gongyuan-pipeline-test.cloud.goog/ e2e, with manually fixed mlmd grpc configmap in user namespaces

### CODE REVIEW

File: frontend/server/app.test.ts [hunk 1]
```diff
@@ -626,23 +626,314 @@ describe('UIServer apis', () => {
...
+      it('does not crash with a weird query', done => {
+        k8sGetCustomObjectSpy.mockImplementation(() =>
+          Promise.resolve(newGetTensorboardResponse()),
+        );
+        // The special case is that, decodeURIComponent('%2') throws an
+        // exception, so this can verify handler doesn't do extra
+        // decodeURIComponent on queries.
+        const weirdLogDir = encodeURIComponent('%2');
+        request.get(`/apps/tensorboard?logdir=${weirdLogDir}&namespace=test-ns`).expect(200, done);
+      });
```
Comment (Bobgy) [inline frontend/server/app.test.ts]: @eterna2 I think the confusion you found with `encodeURIComponent` and `decodeURIComponent` can be demonstrated by this unit test.

This was initially failing with decodeURIComponent throwing on invalid code '%2'. After I removed all the `decodeURIComponent` calls in node server, it's passing. Therefore, the problem was that express automatically parses `req.query.xxx` for us, so we don't need to decode it by ourselves each time. However, most node server handlers decoded for an extra time, so our UI had to encode an extra time too to let these work together.

Now, the fix is clear, we should let node server not call `decodeURIComponent` manually and let UI only encode for once.

Comment (eterna2) [reply frontend/server/app.test.ts]: Do u want me to raise a PR to fix this? Or r u fixing it?

Comment (Bobgy) [reply frontend/server/app.test.ts]: Thanks! I've fixed it for tensorboard handlers.
I will create an issue to track related fixes, just FYI to keep you in the loop.

Comment (Bobgy) [reply frontend/server/app.test.ts]: And if you'd like to provide help, that will definitely welcome!

Comment (Bobgy) [reply frontend/server/app.test.ts]: Created issue: https://github.com/kubeflow/pipelines/issues/3362

**Pull request merged**